packages/notebook/src/browser/contributions/notebook-context-keys.ts

```diff
@@ -0,0 +1,101 @@
+export const NOTEBOOK_CELL_EDITOR_FOCUSED = 'notebookCellEditorFocused';
+export const NOTEBOOK_CELL_MARKDOWN_EDIT_MODE = 'notebookCellMarkdownEditMode';
+export const NOTEBOOK_CELL_LINE_NUMBERS = 'notebookCellLineNumbers';
+export const NOTEBOOK_CELL_EXECUTION_STATE = 'notebookCellExecutionState';
+export const NOTEBOOK_CELL_EXECUTING = 'notebookCellExecuting';
+export const NOTEBOOK_CELL_HAS_OUTPUTS = 'notebookCellHasOutputs';
+export const NOTEBOOK_CELL_INPUT_COLLAPSED = 'notebookCellInputIsCollapsed';
+export const NOTEBOOK_CELL_OUTPUT_COLLAPSED = 'notebookCellOutputIsCollapsed';
+export const NOTEBOOK_CELL_RESOURCE = 'notebookCellResource';
+export const NOTEBOOK_KERNEL = 'notebookKernel';
+export const NOTEBOOK_KERNEL_COUNT = 'notebookKernelCount';
+export const NOTEBOOK_KERNEL_SOURCE_COUNT = 'notebookKernelSourceCount';
+export const NOTEBOOK_KERNEL_SELECTED = 'notebookKernelSelected';
+export const NOTEBOOK_INTERRUPTIBLE_KERNEL = 'notebookInterruptibleKernel';
+export const NOTEBOOK_MISSING_KERNEL_EXTENSION = 'notebookMissingKernelExtension';
+export const NOTEBOOK_HAS_OUTPUTS = 'notebookHasOutputs';
+
+@injectable()
+export class NotebookContextKeyService {
+    createContextKeys(service: ContextKeyService): void {
+        service.createKey(HAS_OPENED_NOTEBOOK, false);
+        service.createKey(KEYBINDING_CONTEXT_NOTEBOOK_FIND_WIDGET_FOCUSED, false);
+
+        // // Is Notebook
+        // export const NOTEBOOK_IS_ACTIVE_EDITOR = ContextKeyExpr.equals('activeEditor', NOTEBOOK_EDITOR_ID);
+        // export const INTERACTIVE_WINDOW_IS_ACTIVE_EDITOR = ContextKeyExpr.equals('activeEditor', INTERACTIVE_WINDOW_EDITOR_ID);
+
+        // Editor keys
+        service.createKey(NOTEBOOK_EDITOR_FOCUSED, false);
+        service.createKey(NOTEBOOK_CELL_LIST_FOCUSED, false);
+        service.createKey(NOTEBOOK_OUTPUT_FOCUSED, false);
+        service.createKey(NOTEBOOK_EDITOR_EDITABLE, true);
+        service.createKey(NOTEBOOK_HAS_RUNNING_CELL, false);
+        service.createKey(NOTEBOOK_USE_CONSOLIDATED_OUTPUT_BUTTON, false);
+        service.createKey(NOTEBOOK_BREAKPOINT_MARGIN_ACTIVE, false);
+        service.createKey(NOTEBOOK_CELL_TOOLBAR_LOCATION, 'left');
+        service.createKey(NOTEBOOK_CURSOR_NAVIGATION_MODE, false);
+        service.createKey(NOTEBOOK_LAST_CELL_FAILED, false);
+
+        // Cell keys
+        service.createKey(NOTEBOOK_VIEW_TYPE, undefined);
+        service.createKey(NOTEBOOK_CELL_TYPE, undefined);
+        service.createKey(NOTEBOOK_CELL_EDITABLE, false);
+        service.createKey(NOTEBOOK_CELL_FOCUSED, false);
+        service.createKey(NOTEBOOK_CELL_EDITOR_FOCUSED, false);
+        service.createKey(NOTEBOOK_CELL_MARKDOWN_EDIT_MODE, false);
+        service.createKey(NOTEBOOK_CELL_LINE_NUMBERS, 'inherit');
+        service.createKey(NOTEBOOK_CELL_EXECUTION_STATE, undefined);
+        service.createKey(NOTEBOOK_CELL_EXECUTING, false); // This only exists to simplify a context key expression, see #129625
```

Comment:
Minor: If you want to keep the reference to the vscode issue, you should link it as a full URL.